test/SILOptimizer/existential_specializer_indirect_class.sil

```diff
@@ -8,74 +8,68 @@ protocol ClassProtocol: AnyObject { func method() }
 
 class C: ClassProtocol { func method() {} }
 
-// CHECK-LABEL: sil shared @$s28test_indirect_class_protocolTf4e_n : $@convention(thin) <τ_0_0 where τ_0_0 : ClassProtocol> (@in τ_0_0) -> ()
-sil hidden @test_indirect_class_protocol : $@convention(thin) (@in ClassProtocol) -> () {
+// CHECK-LABEL: sil [signature_optimized_thunk] [always_inline] @test_indirect_class_protocol : $@convention(thin) (@in ClassProtocol) -> ()
+sil @test_indirect_class_protocol : $@convention(thin) (@in ClassProtocol) -> () {
 // CHECK-NEXT: //
-// CHECK-NEXT: bb0(%0 : $*τ_0_0):
-// CHECK-NEXT:   %1 = load %0
-// CHECK-NEXT:   %2 = init_existential_ref %1
-// CHECK-NEXT:   %3 = alloc_stack $ClassProtocol
-// CHECK-NEXT:   store %2 to %3
+// CHECK-NEXT: bb0(%0 : $*ClassProtocol):
 bb0(%0 : $*ClassProtocol):
-  // CHECK-NEXT: destroy_addr %3
+  // CHECK-NEXT: %1 = load %0
+  // CHECK-NEXT: strong_release %1
+  // CHECK-NEXT: strong_release %1
```

Comment:
Can you understand why this function isn't specialized anymore?

A single `destroy_addr` is now replaced with two releases? Maybe I'm missing something be that doesn't look right.

Reply:
It actually is being specialized. You can see that because the inlined version is different. Anyway, if I disable dead-function-elimination the specialized functions come back. I'll do that.

Reply:
> A single destroy_addr is now replaced with two releases? Maybe I'm missing something be that doesn't look right.

Agreed. It's wrong. The reason for this is in `createApplyWithConcreteType` a new destroy addr instruction is created for each element in `concreteArgCopies`. These are promoted to be `strong_release`s but, a `strong_retain` is never created. If I create a `strong_retain` in `ConcreteArgumentCopy::generate` everything seems to go back to normal but, I don't think those instructions are comparable. Maybe the `destroy_addr` instruction could be replaced with a `strong_release`? Also, how did this work with `copy_addr`?

I really appreciate your patience and explanation. Sorry for all the back and forth we have had (and I'm sure will have). I don't know much about swift's reference counting (I'm eager to learn more, though).